### PR TITLE
Check empty cluster in ci-op's config

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-1010

/hold

require https://github.com/openshift/release/pull/7951

### CODE REVIEW

File: cmd/ci-operator-checkconfig/main.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ func main() {
 
 	seen := map[api.ImageStreamTagReference][]*config.Info{}
 	if err := config.OperateOnCIOperatorConfigDir(configDir, func(configuration *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
+		if err := validateCluster(configuration); err != nil {
```
Comment (stevekuznetsov) [inline cmd/ci-operator-checkconfig/main.go]: This should be a validation that we run on config load, not only here in `checkconfig`

Comment (hongkailiu) [reply cmd/ci-operator-checkconfig/main.go]: I bet it will break our integration tests.
Let me try.

File: test/ci-operator-integration/multi-stage/run.sh [hunk 2]
```diff
@@ -36,7 +36,7 @@ check() {
 }
 
 echo "[INFO] Running ci-operator in dry-mode..."
-if ! ci-operator --dry-run --determinize-output --namespace "${TEST_NAMESPACE}" --config "${TEST_CONFIG0}" 2> "${ERR}" | jq --sort-keys . > "${OUT}"; then
+if ! ci-operator --dry-run --kubeconfig=/Users/hongkliu/.kube/bk.config --determinize-output --namespace "${TEST_NAMESPACE}" --config "${TEST_CONFIG0}" 2> "${ERR}" | jq --sort-keys . > "${OUT}"; then
```
Comment (hongkailiu) [inline test/ci-operator-integration/multi-stage/run.sh]: Need a secret for this.
Will be another PR from release repo.

File: pkg/steps/input_image_tag.go [hunk 3]
```diff
@@ -85,9 +89,13 @@ func (s *inputImageTagStep) Run(ctx context.Context, dry bool) error {
 	}
 
 	if len(s.config.BaseImage.Cluster) > 0 && s.srcClient != s.dstClient {
-		from, err := istObjectReference(s.srcClient, s.config.BaseImage)
-		if err != nil {
-			return fmt.Errorf("failed to reference source image stream tag: %v", err)
+		from := coreapi.ObjectReference{Kind: "DockerImage", Name: fmt.Sprintf("registry.svc.ci.openshift.org/%s/%s@sha256:SHA",
```
Comment (stevekuznetsov) [inline pkg/steps/input_image_tag.go]: This is ugly but @petr-muller  @alvaroaleman I'm not sure we lose anything, really, from this.

File: pkg/api/config.go [hunk 4]
```diff
@@ -114,6 +114,10 @@ func validateBuildRootImageConfiguration(fieldRoot string, input *BuildRootImage
 		return nil
 	}
 
+	if input.ImageStreamTagReference != nil && len(input.ImageStreamTagReference.Cluster) == 0 {
+		return []error{fmt.Errorf("%s.image_stream_tag: no cluster defined", fieldRoot)}
```
Comment (alvaroaleman) [inline pkg/api/config.go]: Its completely pointless that `validateBuildRootImageConfiguration` returns a slice of error rather than a single error.

Can we define a `var errs []error` and append all errors we find rather than returning them?

File: pkg/steps/input_image_tag.go (hunk 3 above)
Comment (alvaroaleman) [reply pkg/steps/input_image_tag.go]: That doesn't really bother me

**Comment by hongkailiu** [comment]
> @hongkailiu can we also put the image registry into a const rather than manually defining it everywhere to make our lives easier should we ever want to change it?

```
grep -irn registry.svc.ci.openshift.org . | grep -v "/test/"  | grep -v "\.md" | grep -v "_test.go"
./cmd/ci-operator/main.go:101:      registry.svc.ci.openshift.org/ci-op-9o8bacu/stable:${component}
./pkg/webreg/webreg.go:632:FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.13 AS builder
./pkg/webreg/webreg.go:641:FROM registry.svc.ci.openshift.org/openshift/origin-v4.5:base
./pkg/webreg/webreg.go:652:      - "registry.svc.ci.openshift.org/ocp/builder:golang-1.13"
./pkg/steps/multi_stage.go:319:         Image:                    "registry.svc.ci.openshift.org/ci/secret-wrapper:latest",
./pkg/steps/input_image_tag.go:92:              from := coreapi.ObjectReference{Kind: "DockerImage", Name: fmt.Sprintf("registry.svc.ci.openshift.org/%s/%s@sha256:SHA",
./pkg/steps/clusterinstall/template.go:149:        value: registry.svc.ci.openshift.org/${NAMESPACE}/release
./pkg/steps/clusterinstall/template.go:382:        value: registry.svc.ci.openshift.org/${NAMESPACE}/release

```

Except test files and being a part of multi-line string, this is the only ocurrence:

https://github.com/openshift/ci-tools/blob/5a160a8c8ec3f6d3cd7b97d86b09ebc84a8ff298/pkg/steps/multi_stage.go#L326

Do you want me to define a constant and use it in those 2 places?

**Comment by alvaroaleman** [comment]
> Do you want me to define a constant and use it in those 2 places?

yes, please

File: pkg/api/config.go (hunk 4 above)
Comment (hongkailiu) [reply pkg/api/config.go]: Done.
